gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjureJavaLocalCodegenPlugin.java

```diff
@@ -0,0 +1,128 @@
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.conjure;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import com.palantir.gradle.conjure.api.ConjureExtension;
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.Dependency;
+import org.gradle.api.plugins.BasePlugin;
+import org.gradle.api.plugins.JavaLibraryPlugin;
+import org.gradle.api.tasks.Copy;
+import org.gradle.api.tasks.TaskProvider;
+
+public final class ConjureJavaLocalCodegenPlugin implements Plugin<Project> {
+    private static final String CONJURE_CONFIGURATION = "conjure";
+    private static final Pattern DEFINITION_NAME =
+            Pattern.compile("(.*)-([0-9]+\\.[0-9]+\\.[0-9]+(?:-rc[0-9]+)?(?:-[0-9]+-g[a-f0-9]+)?)(\\.conjure)?\\.json");
+
+    static final String CONJURE_JAVA = "conjureJava";
+    static final String CONJURE_JAVA_BINARY = "com.palantir.conjure.java:conjure-java";
+
+    @Override
+    public void apply(Project project) {
+        project.getPlugins().apply(BasePlugin.class);
+
+        ConjureExtension extension =
+                project.getExtensions().create(ConjureExtension.EXTENSION_NAME, ConjureExtension.class);
+
+        Configuration conjureIrConfiguration = project.getConfigurations().create(CONJURE_CONFIGURATION);
+        TaskProvider<Copy> extractConjureIr = project.getTasks().register("extractConjureIr", Copy.class, task -> {
+            task.rename(DEFINITION_NAME, "$1.conjure.json");
+            task.from(conjureIrConfiguration);
+            task.into(project.getLayout().getBuildDirectory().dir("conjure-ir"));
+        });
+
+        Configuration conjureJavaConfig = project.getConfigurations().maybeCreate(CONJURE_JAVA);
+        File conjureJavaDir = new File(project.getBuildDir(), CONJURE_JAVA);
+        project.getDependencies().add(CONJURE_JAVA, CONJURE_JAVA_BINARY);
+        ExtractExecutableTask extractJavaTask = ExtractExecutableTask.createExtractTask(
+                project, "extractConjureJava", conjureJavaConfig, conjureJavaDir, "conjure-java");
+
+        setupSubprojects(project, extension, extractJavaTask, extractConjureIr, conjureIrConfiguration);
+    }
+
+    private static void setupSubprojects(
+            Project project,
+            ConjureExtension extension,
+            ExtractExecutableTask extractJavaTask,
+            TaskProvider<Copy> extractConjureIr,
+            Configuration conjureIrConfiguration) {
+
+        // Validating that each subproject has a corresponding definition and vice versa.
+        // We do this in afterEvaluate to ensure the configuration is populated.
+        project.afterEvaluate(p -> {
+            Set<String> apis = conjureIrConfiguration.getAllDependencies().stream()
```

Comment:
`getIncoming().getDependencies()` as it ensures that dependency actions are run too (e.g. default dependencies)

Reply:
```suggestion
            Set<String> apis = conjureIrConfiguration.getIncoming().getDependencies().stream()
```